tools/SourceKit/lib/Core/Context.cpp

```diff
@@ -16,9 +16,11 @@
 
 using namespace SourceKit;
 
-SourceKit::Context::Context(StringRef RuntimeLibPath)
+SourceKit::Context::Context(StringRef RuntimeLibPath,
+    		llvm::function_ref<
+    			std::unique_ptr<LangSupport>(Context &)> LangSupportFactoryFn)
   : RuntimeLibPath(RuntimeLibPath),
-    SwiftLang(LangSupport::createSwiftLangSupport(*this)),
+    SwiftLang(LangSupportFactoryFn(*this)),
     NotificationCtr(new NotificationCenter()) {
 }
 
```

Comment:
I recommend that the factory function is called last, after `Context` is fully initialized. For example, the factory function may want to access the `NotificationCtr` variable. So change to something like this:
```
   : RuntimeLibPath(RuntimeLibPath),
      NotificationCtr(new NotificationCenter()) {
    // Should be called last after everything is initialized.
    SwiftLang = LangSupportFactoryFn(*this);
  }